### DISCUSSION TITLE
Poll Phase Infinity ?

### DISCUSSION THREAD

**Discussion by hazooma** [libuv/libuv#3288] [Q&A]
Hello , 
I have a bit confusion a bout polling phase for libuv, as i understand the timeout is calculated before ahead and then Poll phase in entered and the thread will be blocked for that time out ? right ? 
what will happen if in some case the Polling phase time out was decided to be Infinity and then event loop entered the poll phase with that timeout and then suddenly a  setTimeout operation occurred ? what will happen in that case ? the event-loop will remain in the poll phase ?

**Comment by santigimeno** [comment]
Being single-threaded, that scenario cannot happen. If the calculated `timeout` was set to infinity, no new code will run until poll returns, so no `setTimeout()` could happen.

**Reply by hazooma** [reply]
> Being single-threaded, that scenario cannot happen. If the calculated `timeout` was set to infinity, no new code will run until poll returns, so no `setTimeout()` could happen.

I see, but isn't that bad ? like that single thread will be blocked all that time till Poll phase return ?
